engine/table/src/main/java/io/deephaven/engine/table/impl/select/analyzers/SelectAndViewAnalyzer.java

```diff
@@ -521,17 +655,76 @@ public void close() {
      *        {@link io.deephaven.engine.liveness.LivenessReferent liveness referents}
      * @param onCompletion Called when an inner column is complete. The outer layer should pass the {@code onCompletion}
      */
-    public abstract void applyUpdate(TableUpdate upstream, RowSet toClear, UpdateHelper helper,
-            JobScheduler jobScheduler, @Nullable LivenessNode liveResultOwner,
-            SelectLayerCompletionHandler onCompletion);
+    public void applyUpdate(
+            final TableUpdate upstream,
+            final RowSet toClear,
+            final UpdateHelper helper,
+            final JobScheduler jobScheduler,
+            @Nullable final LivenessNode liveResultOwner,
+            final Layer.CompletionHandler onCompletion) {
+
+        TableUpdateImpl postFlatten = null;
+        final MutableObject<TableUpdateImpl> postFlattenHolder = new MutableObject<>();
+        final Layer.CompletionHandler[] handlers = new Layer.CompletionHandler[layers.size()];
+
+        final BitSet allLayers = new BitSet();
+        setAllNewColumns(allLayers);
+
+        final Layer.CompletionHandler innerHandler = new Layer.CompletionHandler(
+                allLayers, onCompletion) {
+            @Override
+            protected void onError(Exception error) {
+                // propagate the error upstream
+                onCompletion.onError(error);
+            }
+
+            @Override
+            void onLayerCompleted(int completedColumn) {
+                super.onLayerCompleted(completedColumn);
+
+                for (int ii = 1; ii < layers.size(); ++ii) {
+                    handlers[ii].onLayerCompleted(completedColumn);
+                }
+
+                onCompletion.onLayerCompleted(completedColumn);
+            }
+
+            @Override
+            protected void onAllRequiredColumnsCompleted() {
+                final TableUpdateImpl update = postFlattenHolder.getValue();
+                if (update != null) {
+                    update.release();
+                }
+            }
+        };
+
+        for (int ii = layers.size() - 1; ii >= 0; --ii) {
+            final Layer currentLayer = layers.get(ii);
+            handlers[ii] = currentLayer.createUpdateHandler(
+                    postFlatten != null ? postFlatten : upstream,
+                    toClear, helper, jobScheduler, liveResultOwner, innerHandler);
+
+            if (currentLayer instanceof StaticFlattenLayer) {
+                postFlatten = new TableUpdateImpl();
+                postFlatten.added = ((StaticFlattenLayer) currentLayer).getParentRowSetCopy();
```

Comment:
We decided this was silly and overcomplicating things. We can always send the parent's row set in from outside, and flatten before the first layer that needs it (or just make the layers independent and know what they want).